test/index.js

```diff
@@ -451,4 +473,79 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('Manifest v3 changes', function () {
+    it('can still verify password', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to verify the password
+      const result = await keyringController.verifyPassword(password);
+      expect(result).toBeTruthy();
+    });
+
+    it('can still unlock with password after being migrated and locked', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to unlock the keychaing via old password
+      const result = await keyringController.submitPassword(password);
+      expect(result).toBeTruthy();
+
+      // Ensure the new vault value has separator and salt
+      const { vault, salt } = keyringController.store.getState();
+      expect(vault).toContain(':::');
+      expect(salt).toBeTruthy();
+    });
+
+    // TODO:  THIS SHOULD BE FAILING
+    it('does not unlock with incorrect encryption key', async function () {
+      await keyringController.submitPassword(password);
+
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to login with an incorrect encryption key
+      const encryptionKey = await keyringController.submitEncryptedKey(
+        'FAKE KEY',
+      );
+
+      expect(encryptionKey).rejects.toThrow();
```

Comment:
This test is still passing, which it shouldn't.  I think this is affected by the `_generateEncryptedKey` returning what looks like an empty string in the just the tests.

Reply:
There's no way this test could work. 
You're using an async assertion which should be getting a promise as input, but you await that promise above and pass the return value.

Avoid awaiting the  submitEncryptedKey call, assign the promise it returns to a variable and/or put that reference inside expect(_).rejects